# MASTERS_manuscript
Code and relevant items to accompany Muscle Transcriptional Networks Linked to Resistance Exercise Training Hypertrophic Response Heterogeneity‚Äù (Lavin &amp; Bell, et al., Physiol Genomics submission). 
This code describes application of network biology tools WGCNA and PLIER to phenotypic exercise data.

## Background and Objectives
This project aims to identify transcriptome-level networks associated with positive changes in skeletal muscle mass following exposure to exercise training. 

Networks were interrogated in three analyses:
1. Prediction: networks whose baseline expression is associated with a positive outcome following resistance exercise training
2. Response: networks that change following exposure to resistance exercise training, independently of degree of muscle hypertrophy
3. Plasticity: networks that change following exposure to resistance exercise training in a manner concordant with degree of muscle hypertrophy

The pipeline was designed to optimize the detection of all three network types using two computational tools: Weighted Gene Correlation Network Analysis (WGCNA)and Pathway-Level Information Extractor (PLIER). The relative benefits and drawbacks of each are available in the documentation and accompanying manuscript by Lavin & Bell, et al. 

This pipeline is currently designed for gene expression data but can be applied to other omics. Users are encouraged to curate relevant pathway associations for molecules of interest (e.g., metabolites, peptides) in order to optimize use of PLIER. WGCNA is an agnostic tool that has already been applied to other 'omics with success. 

## Sub-directories
### 0_data
Contains input data including gene-pathway association matrices developed from:
- Meta Analysis of Muscle Exercise (Pillon, et al., 2020, https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6981202/)
- Muscle Fiber-Type Markers (Rubenstein, et al., 2020, https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6959232/)
- Links to download gene expression data used in the current study from NIH GEO.

### 1_code
Contains 
- Code for WGCNA and PLIER analyses
- An HTML document compiled using knitr from R markdown. 

### 2_pipeline
Manuscript figure summarizing the workflow.

### 3_output
List of variables that are generated by running this pipeline. 

### 4_functions
User-defined functions to aid in preparation of figures and tables for manuscript preparation.

## Acknowledgments
The primary pipelines used and described herein were developed by expert bioinformaticians. Please consult the original materials for continued reading and further exploration of the intricacies of these algorithms:

### Weighted Gene Correlation Network Analysis
- Peter Langfelder and Steve Horvath
- Tutorials and other materials: https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/index.html

### Pathway-Level Information Extractor
- Weiguang Mao and Maria Chikina
- Github repository: https://github.com/wgmao/PLIER
